app/controllers/ui_ansible_roles_controller.rb

```diff
@@ -3,12 +3,34 @@ def resource_name(resource = 'AnsibleRole')
     super resource
   end
 
+  layout 'api/v2/layouts/index_layout', :only => [:index, :variables]
+
   def index
     @ui_ansible_roles = resource_scope_for_index(:permission => :view_ansible_roles)
   end
 
+  def variables
+    @ui_ansible_roles = resource_scope(:permission => :view_ansible_roles).search_for(*search_options).includes(:ansible_variables)
```

Comment:
I prefer `preload` if you wouldn't mind - it is more clear what we want to achieve. Or is the assoc needed in the search?